lib/codecs/src/encoding/format/raw_message.rs

```diff
@@ -49,16 +54,16 @@ impl Encoder<Event> for RawMessageSerializer {
     type Error = vector_common::Error;
 
     fn encode(&mut self, event: Event, buffer: &mut BytesMut) -> Result<(), Self::Error> {
-        let message_key = log_schema().message_key();
-
-        let log = event.as_log();
-
-        if let Some(bytes) = log
-            .get_by_meaning(message_key)
-            .or_else(|| log.get(message_key))
-            .map(|value| value.coerce_to_bytes())
-        {
-            buffer.put(bytes);
+        if let Some(message_key) = log_schema().message_key() {
+            let log = event.as_log();
+
+            if let Some(bytes) = log
+                .get_by_meaning(message_key.to_string().as_str())
```

Comment:
This seems line an expensive operation (`message_key.to_string()`) to be doing on each event encoding. Could it be amortized elsewhere?

Reply:
Excellent catch. We can improve the existing code and avoid expensive copies by using [get_message()](https://github.com/vectordotdev/vector/blob/master/lib/vector-core/src/event/log_event.rs#L478). This is efficient both Vector and Legacy namespaces.